airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,182 @@
+import { Field, FieldInputProps, FieldProps, FormikProps } from "formik";
+import { ChangeEvent, useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { ControlLabels, DropDown, DropDownRow, Input } from "components";
+
+import { ConnectionScheduleData, ConnectionScheduleType } from "core/request/AirbyteClient";
+
+import availableCronTimeZones from "../../../../config/availableCronTimeZones.json";
+import { ConnectionFormMode } from "../ConnectionForm";
+import { FormikConnectionFormValues, useFrequencyDropdownData } from "../formConfig";
+import styles from "./ScheduleField.module.scss";
+
+interface ScheduleFieldProps {
+  scheduleData: ConnectionScheduleData | undefined;
+  mode: ConnectionFormMode;
+  onDropDownSelect?: (item: DropDownRow.IDataItem) => void;
+}
+
+const CRON_DEFAULT_VALUE = {
+  cronTimeZone: "UTC",
+  // Fire at 12:00 PM (noon) every day
+  cronExpression: "0 0 12 * * ?",
+};
+
+const ScheduleField: React.FC<ScheduleFieldProps> = ({ scheduleData, mode, onDropDownSelect }) => {
+  const { formatMessage } = useIntl();
+  const frequencies = useFrequencyDropdownData(scheduleData);
+
+  const onScheduleChange = (item: DropDownRow.IDataItem, form: FormikProps<FormikConnectionFormValues>) => {
+    onDropDownSelect?.(item);
+
+    let scheduleData: ConnectionScheduleData;
+    const isManual = item.value === ConnectionScheduleType.manual;
+    const isCron = item.value === ConnectionScheduleType.cron;
+
+    // Set scheduleType for yup validation
+    const scheduleType = isManual || isCron ? (item.value as ConnectionScheduleType) : ConnectionScheduleType.basic;
```

Comment:
I see we're defaulting to manual versus null. Could this have unintended consequences or is this just explicitly setting what was implicit before?

Reply:
Yeah we didn't have this field(scheduleType) before, this just does that job which was happening by setting scheduleData to null in a more implicit way

Reply:
Okay, just making sure. We need to ensure if we have scheduleData we have a scheduleType but I think it's covered!